services/account-without-juggler/repositories/account/datasources/mysqlconn.ts

```diff
@@ -0,0 +1,229 @@
+const debug = require("debug")("accountsansjuggler");
+const mysql = require("mysql");
+import {
+  CrudConnector,
+  EntityData
+} from "loopback-next/packages/repository/src/crud-connector";
+import {
+  Class,
+  DataSource,
+  Entity,
+  Filter,
+  ObjectType,
+  Options,
+  Where
+} from "loopback-next/packages/repository";
+
+export class MySqlConn implements CrudConnector {
+  private connection: any;
+
+  constructor(config: Object) {
+    this.connection = mysql.createConnection(config);
+    if (this.connection.state !== "authenticated") {
+      this.connect().then(
+        function(res) {
+          debug("Connection success ", res);
+        },
+        function(err) {
+          debug("err connecting", err);
+        }
+      );
+    }
+  }
+  name: "mysql";
+  interfaces?: string[];
+  connect() {
+    let self = this;
+    return new Promise<void>(function(resolve, reject) {
+      self.connection.connect(function(err: any) {
```

Comment:
Have you considered using https://www.npmjs.com/package/mysql-promise or https://www.npmjs.com/package/promise-mysql, instead of this manual way of adding promise support?